api/src/migrations/geoprocessing/1611221157285-initialGeoDBSetup.ts

```diff
@@ -0,0 +1,185 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+
+export class initialGeoDBSetup1611221157285 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+      await queryRunner.query(`
+      CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+      CREATE EXTENSION IF NOT EXISTS tablefunc;
+      CREATE EXTENSION IF NOT EXISTS plpgsql;
+      CREATE EXTENSION IF NOT EXISTS postgis;
+      CREATE EXTENSION IF NOT EXISTS postgis_raster; -- OPTIONAL
+      CREATE EXTENSION IF NOT EXISTS postgis_topology; -- OPTIONAL
+
+      CREATE TYPE "source_type" AS ENUM (
+        'user_imported',
+        'gbif',
+        'uicn',
+        'ecoregions',
+        'intersection'
+      );
+
+      CREATE TYPE "shape_type" AS ENUM (
+        'square',
+        'hexagon',
+        'irregular'
+      );
+
+      CREATE TYPE "status" AS ENUM (
+        'created',
+        'running',
+        'done',
+        'failure'
+      );
+
+      CREATE TYPE "adm_level" AS ENUM (
+        0,
+        1,
+        2
+      );
+
+-- Administrative regions table.
+      CREATE TABLE "admin_regions" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "ogc_fid" int NOT NULL,
+        "the_geom" geometry CONSTRAINT admin_regions_geometry_valid_check CHECK (ST_IsValid(the_geom)),
+        "name_0" varchar,
+        "name_1" varchar,
+        "name_2" varchar,
+        "iso3" varchar(3),
+        "gid_0" varchar,
+        "gid_1" varchar,
+        "gid_2" varchar,
+        "level" adm_level NOT NULL,
```

Comment:
as big fan as I am of enums, couldn't we just use `admin_level integer CHECK (admin_level >= 0 AND admin_level <=2)`?

Reply:
I didn't check the syntax, please take it as pseudocode just to show the intention 😬

Reply:
I think numeric enums won't work in any case, so either the enum is

```CREATE TYPE "adm_level" AS ENUM (
        '0',
        '1',
        '2'
      );
```

or we use the check on integers I suggested above (I still think it's the simplest option!)